lib/html_parser.dart

```diff
@@ -747,11 +748,16 @@ class HtmlParser extends StatelessWidget {
   static StyledElement _removeEmptyElements(StyledElement tree) {
     List<StyledElement> toRemove = <StyledElement>[];
     bool lastChildBlock = true;
-    tree.children.forEach((child) {
+    tree.children.forEachIndexed((index, child) {
       if (child is EmptyContentElement || child is EmptyLayoutElement) {
         toRemove.add(child);
       } else if (child is TextContentElement
-          && (tree.name == "body" || tree.name == "ul")
```

Comment:
This code is getting s complex... I am sure at one point we need to rewrite this 'node cleanup' altogether.